packages/compass-e2e-tests/tests/collection-aggregations-tab.test.ts

```diff
@@ -497,6 +497,153 @@ describe('Collection aggregations tab', function () {
     });
   });
 
+  async function runAggregation() {
+    await browser.clickVisible(Selectors.RunPipelineButton);
+    const resultsWorkspace = await browser.$(
+      Selectors.AggregationResultsWorkspace
+    );
+    await resultsWorkspace.waitForDisplayed();
+  }
+
+  async function editPipeline() {
+    await browser.clickVisible(Selectors.EditPipelineButton);
+    const builderWorkspace = await browser.$(
+      Selectors.AggregationBuilderWorkspace
+    );
+    await builderWorkspace.waitForDisplayed();
+  }
+
+  async function getDocuments() {
+    // Switch to JSON view so it's easier to get document value
+    await browser.clickVisible(
+      Selectors.AggregationResultsJSONListSwitchButton
+    );
+    // Get all visible documents
+    const documents = await browser.$$(Selectors.DocumentJSONEntry);
+    // Get ace editor content and parse it
+    const parsed = await Promise.all(
+      documents.map(async (doc) => {
+        const aceEditor = await doc.$('.ace_content');
+        return JSON.parse(await aceEditor.getText());
+      })
+    );
+    return parsed;
+  }
+
+  it('supports running and editing aggregation', async function () {
+    // Set first stage to match
+    await browser.focusStageOperator(0);
+    await browser.selectStageOperator(0, '$match');
+    await browser.setAceValue(Selectors.stageEditor(0), '{ i: 5 }');
+
+    // Run and wait for results
+    await runAggregation();
+
+    // Get all documents from the current results page
+    const docs = await getDocuments();
+
+    expect(docs).to.have.lengthOf(1);
+    expect(docs[0]).to.have.property('_id');
+    expect(docs[0]).to.have.property('i', 5);
+    expect(docs[0]).to.have.property('j', 0);
+
+    // Go back to the pipeline builder
+    await editPipeline();
+
+    // Change match filter
+    await browser.setAceValue(
+      Selectors.stageEditor(0),
+      '{ i: { $gte: 5, $lte: 10 } }'
+    );
+
+    // Run and wait for results
+    await runAggregation();
+
+    // Get all documents from the current results page
+    const updatedDocs = await getDocuments();
+
+    // Check that the documents are matching pipeline
+    expect(updatedDocs).to.have.lengthOf(6);
+    expect(updatedDocs[0]).to.have.property('i', 5);
+    expect(updatedDocs[1]).to.have.property('i', 6);
+    expect(updatedDocs[2]).to.have.property('i', 7);
+    expect(updatedDocs[3]).to.have.property('i', 8);
+    expect(updatedDocs[4]).to.have.property('i', 9);
+    expect(updatedDocs[5]).to.have.property('i', 10);
+
+    // Go back to the pipeline builder because that's what beforeEach expects
+    await editPipeline();
```

Comment:
One issue with this pattern is that if this test fails, all the next ones will likely fail as well if they expect to start in a specific state. Maybe safer to stick this in the beforeEach?

Reply:
(No idea why this comment doubled up)